docs/doc_4_authentication.md

```diff
@@ -40,3 +39,160 @@ func main() {
+
+## OAuth Authentication
+
+### Authenticating with OAuth ClientID and ClientSecret
+
+### Admin API Authentication using Service Accounts
+
+Authenticating using Service Accounts
+
+```go
+package main
+
+import (
+	"log"
+	"os"
+
+	"go.mongodb.org/atlas-sdk/v20241023001/admin"
+)
+
+func main() {
+	clientID := os.Getenv("MONGODB_ATLAS_CLIENT_ID")
+	clientSecret := os.Getenv("MONGODB_ATLAS_CLIENT_SECRET")
+
+	if clientID == "" || clientSecret == "" {
+		log.Fatal("Missing CLIENT_ID or CLIENT_SECRET environment variables")
+	}
+
+	// Using ClientID and ClientSecret. No cache supported (nil).
+	sdk, err := admin.NewClient(admin.UseOAuthAuth(clientID, clientSecret, nil))
+
+	// Make API calls
+}
+```
+
+> NOTE: This method of initialization provides **No Ability to Revoke API**: once a token is issued, it cannot be invalidated until it expires.
+
+### Specifying a Token Cache
+
+In this example, we will demonstrate how to use the OAuth Client Credentials flow with a custom token cache.
+The cache allows you to store OAuth tokens for reuse across application restarts, improving efficiency by reducing the number of token requests to the authorization server.
+
+**Note:** Cache prevents from requesting multiple tokens which are subject to OAuth Token Limits and Rate Limiting.
+
+```go
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"os"
+	"sync"
+
+	"go.mongodb.org/atlas-sdk/v20241023001/auth/credentials"
+	"go.mongodb.org/atlas-sdk/v20241023001/admin"
+)
+
+type MyTokenCache struct {
+	fileContent []byte
+	mu          sync.Mutex
+}
+
+func (s *MyTokenCache) RetrieveToken(_ context.Context) (string, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	var tkn string
+	err := json.Unmarshal(s.fileContent, &tkn)
+	if err != nil {
+		return "", err
+	}
+
+	return tkn, nil
+}
+
+func (s *MyTokenCache) SaveToken(_ context.Context, tkn string) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	fileData, err := json.Marshal(tkn)
+	if err != nil {
+		return err
+	}
+	s.fileContent = fileData
+	return nil
+}
+
+func main() {
+	clientID := os.Getenv("MONGODB_ATLAS_CLIENT_ID")
+	clientSecret := os.Getenv("MONGODB_ATLAS_CLIENT_SECRET")
+
+	if clientID == "" || clientSecret == "" {
+		log.Fatal("Missing CLIENT_ID or CLIENT_SECRET environment variables")
+	}
+
+	fileTokenCache := MyTokenCache{}
+	// Using ClientID and ClientSecret. No cache supported (nil).
+	httpClient, err := admin.NewClient(admin.UseOAuthAuth(clientID, clientSecret, fileTokenCache))
+	sdk, err := admin.NewClient(admin.UseHTTPClient(httpClient))
+
+	// Make requests to the API
+	// ...
+	// Revoke Token
+	_ = tokenSource.RevokeToken()
+}
+```
+
+## Advanced Use Cases
+
+### Revocation
+
+Revocation invalidates the token before its expiration date.
+Revocation should be used to "logg out" the current OAuth client and configuring a new one.
```

Comment:
```suggestion
Revocation invalidates an OAuth token before its expiration date. This effectively "logs out" the current OAuth client and allows you to configure a new client.
```

Generally, at the start of a new section, it's best to be very explicit (i.e. "an OAuth token" vs. "the token"). An exception to this case is when you are writing a strict sequence of events and "the token" can be unambiguously taken to refer to a specific token created previously in that sequence.

Also, rather than using the suggestive "should", let's stick to telling the user what something does or what is possible, unless we are making explicit recommendations from among alternatives.